### ISSUE TITLE
PHP Strict, if a field on the response or request was removed - results to broken response

### ISSUE THREAD

**Issue by louiemiranda** [frapi/frapi#204]
Just want to know if Frapi really set to throw error for undeclared variables?

<?xml version="1.0" encoding="UTF-8"?>
<response>
    <errors>    <error code="PHP Notice error">
            <message>Undefined index:  errormessage (Error Number: 8), (File: /var/www/file.php at line 127)</message>
            <name>PHP Notice error</name>
            <at></at>
        </error>
    </errors>
</response>

I tried to add this on AllFiles.php
ini_set('display_startup_errors', FALSE);
ini_set('display_errors', FALSE);

Any ideas? Is this normal? Using the last frapi on git

**Comment by trevormorse** [comment]
@louiemiranda Yes - Frapi is set to throw errors on undeclared variables. Can you show me a snippet of the code that is throwing this error so I can investigate further?

**Comment by louiemiranda** [comment]
@trevormorse Ah, that explains it.

I have a different project, using Frapi API to communicate to another API.

Anyway, this is the sample XML that was received by Frapi from a curl request.

```
<?xml version="1.0" encoding="utf-8"?>
<transaction>
    <result>SUCCESS</result>
    <merchanttransid>5653434</merchanttransid>
    <transref>248630</transref>
    <errorcode>000</errorcode>
    <errormessage/>
    <description>Approved</description>
</transaction>
```

Since the errormessage is blank, Frapi throws...

```
<response>
    <errors>    <error code="PHP Notice error">
            <message>Undefined index:  errormessage (Error Number: 8), (File: /var/www/html/src/frapi/custom/Interface/Gateway3.php at line 127)</message>
            <name>PHP Notice error</name>
            <at></at>
        </error>
    </errors>

</response>
```

I saw application.ini and found
- phpSettings.display_startup_errors = 0
- phpSettings.display_errors = 0

Tried to switch it off and all, but I think that is not the place to look at it.

If ever, can the strict option be turned off? I will try to look on the core, but you may have a better idea. The issue is, if a response is empty and valid - Frapi throws in an undefined index error message.

Hope you can help me.

Regards and thanks

**Comment by louiemiranda** [comment]
As an initial fix.

<code>
$response = $this->Gateway->Charge($request);

if (empty($response['errormessage'])) {
    $response['errormessage'] = '';
}
</code>

This worked. But, my worries are... Since, it is coming from a different server API request, I need to know all variables. It may work at times, but will break Frapi if there is a change made without you knowing it.

Any help or leads would be great.

**Comment by trevormorse** [comment]
@louiemiranda error handler for frapi was being set for E_ALL regardless of setting in php.ini. See my commit/pull request for a fix that will read the setting from php.ini, should help you get it working in your case.

**Comment by louiemiranda** [comment]
Thanks @trevormorse this will do the trick.

**Comment by AidenAndrews** [comment]
I had this issue at first when blocking put on one of my routes i fixed it by going into frapi admin -> errors -> edit the error -> Typed a message into http reason phrase
